File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    pocketmon_amount = (len(nums)//2)
    pocketmon_set = set()
    
    for n in nums:
        pocketmon_set.add(n)
    
    if len(pocketmon_set) >= pocketmon_amount:
        return pocketmon_amount
    
    else:
        return len(pocketmon_set)
    

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>

using namespace std;

int solution(vector<int> nums)
{
    unordered_set<int> s(nums.begin(), nums.end());
    int max_take = nums.size() / 2;
    
    return min((int)s.size(), max_take);
}